x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtValidateUtil.java

```diff
@@ -286,17 +280,17 @@ public static void validateSignature(final SignedJWT jwt, final List<JWK> jwks)
         assert jwks != null && jwks.isEmpty() == false : "Caller must provide a non-empty JWK list";
         final String id = jwt.getHeader().getKeyID();
         final JWSAlgorithm alg = jwt.getHeader().getAlgorithm();
-        LOGGER.trace("JWKs [" + jwks.size() + "], JWT KID [ + id + ], and JWT Algorithm [" + alg.getName() + "] before filters.");
+        LOGGER.trace("JWKs [{}], JWT KID [{}], and JWT Algorithm [{}] before filters.", jwks.size(), id, alg.getName());
 
         // If JWT has optional kid header, and realm JWKs have optional kid attribute, any mismatches JWT.kid vs JWK.kid can be ignored.
         // Keep any JWKs if JWK optional kid attribute is missing. Keep all JWKs if JWT optional kid header is missing.
         final List<JWK> jwksKid = jwks.stream().filter(j -> ((id == null) || (j.getKeyID() == null) || (id.equals(j.getKeyID())))).toList();
-        LOGGER.trace("JWKs [" + jwksKid.size() + "] after KID [" + id + "||null] filter.");
+        LOGGER.trace("JWKs [{}] after KID [{}](|null) filter.", jwksKid.size(), id);
```

Comment:
Nit. This is more than a parameterization change, content of the message changed too. It is not clear if that was intended. If intended then OK.
- Before: `KID[123||null]`.
- After: `KID[123](|null)`.